sleap_roots/bases.py

```diff
@@ -25,8 +34,15 @@ def get_root_lengths(pts: np.ndarray) -> np.ndarray:
 
     Returns:
         Array of root lengths of shape (instances,).
+        If there is no root, or the roots is one point only (all of the rest of the
+        points are NaNs), an array of NaNs with shape (len(pts),) is returned.
+        This is also the case for non-contiguous points at the moment.
     """
     segment_diffs = np.diff(pts, axis=1)
     segment_lengths = np.linalg.norm(segment_diffs, axis=-1)
-    total_lengths = np.nansum(segment_lengths, axis=-1)
+
+    if np.isnan(segment_lengths).all():
+        total_lengths = np.empty((len(pts),)) ** np.nan
+    else:
+        total_lengths = np.nansum(segment_lengths, axis=-1)
```

Comment:
I don't think this is the correct behavior.

Consider a case where you pass this function a single root that only has one point, so we can't define the segment length:

```
pts.shape == (1, nodes, 2)
total_lengths == (np.nan,)
```

Now consider a case where the same root is detected together with one that has a positive length. Now, that *exact same root* which previously had a root length of `np.nan` now has a root length of 0 depending on whether we call this function with other roots:

```
pts.shape == (2, nodes, 2)
total_lengths == (0, 50.2)
```

What we should do here instead of having this if statement is to just mask them appropriately:

```py
total_lengths = np.nansum(segment_lengths, axis=-1)
total_lengths[np.isnan(segment_lengths).all(axis=-1)] = np.nan
```

Reply:
Done and the test passes.